IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcBorehole/DocEntity.xml

```diff
@@ -1,14 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcBorehole" UniqueId="3c670ec4-0027-4bd5-a2d2-bff02387ab99" BaseDefinition="IfcGeoScienceElement" EntityFlags="32">
 	<Attributes>
-		<DocAttribute Name="PredefinedType" UniqueId="0bf60023-d937-4efa-b4de-82a645a3355f" DefinedType="IfcBoreholeTypeEnum">
+		<DocAttribute Name="PredefinedType" UniqueId="0bf60023-d937-4efa-b4de-82a645a3355f" DefinedType="IfcBoreholeTypeEnum" AttributeFlags="1">
 			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets.</Documentation>
 		</DocAttribute>
 	</Attributes>
 	<WhereRules>
-		<DocWhereRule Name="HasObjectType" UniqueId="7fc43f81-5064-48a6-8eb2-317cb98893fa">
-			<Documentation>The attribute ObjectType shall be given if the predefined type is set to USERDEFINED.</Documentation>
-			<Expression>(PredefinedType &lt;&gt; IfcBoreholeTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)</Expression>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="74f2eb30-3c94-4216-9492-bc4fb075f42d">
+			<Documentation>Either the _PredefinedType_ attribute is unset(e.g. because an _IfcBoreholeType_ is associated), or the inherited attribute _ObjectType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
```

Comment:
```suggestion
			<Documentation>Either the _PredefinedType_ attribute is unset (e.g. because an _IfcBoreholeType_ is associated), or the inherited attribute _ObjectType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
```